src/main/java/com/amazonaws/encryptionsdk/model/CiphertextHeaders.java

```diff
@@ -179,6 +179,9 @@ public Boolean isComplete() {
      */
     private int parseVersion(final byte[] b, final int off) throws ParseException {
         version_ = PrimitivesParser.parseByte(b, off);
+        if(CiphertextType.deserialize(version_) == null) {
```

Comment:
This is not the correct check.

We currently only have one valid version: `1`. Any other value MUST throw an error.